com.unity.render-pipelines.core/Runtime/Debugging/DebugManager.cs

```diff
@@ -264,20 +280,34 @@ void EnsurePersistentCanvas()
             }
         }
 
-        internal void TogglePersistent(DebugUI.Widget widget)
+        internal void TogglePersistent(DebugUI.Widget widget, int? forceTupleIndex = null)
         {
             if (widget == null)
                 return;
 
-            var valueWidget = widget as DebugUI.Value;
-            if (valueWidget == null)
+            EnsurePersistentCanvas();
+
+            switch (widget)
             {
-                Debug.Log("Only DebugUI.Value items can be made persistent.");
-                return;
+                case DebugUI.Value value:
+                    m_RootUIPersistentCanvas.Toggle(value);
+                    break;
+                case DebugUI.ValueTuple valueTuple:
+                    m_RootUIPersistentCanvas.Toggle(valueTuple, forceTupleIndex);
+                    break;
+                case DebugUI.Container container:
+                    // When container is toggled, we make sure that if there are ValueTuples, they all get the same element index.
+                    int pinnedIndex = container.children.Max(w => (w as DebugUI.ValueTuple)?.pinnedElementIndex ?? -1);
+                    foreach (var child in container.children)
+                    {
+                        if (child is DebugUI.Value || child is DebugUI.ValueTuple)
+                            TogglePersistent(child, pinnedIndex);
+                    }
+                    break;
+                default:
+                    Debug.Log("Only readonly items can be made persistent.");
```

Comment:
should we keep the debug log here? i assumed this will show up in console and i am not sure what the users are supposed to do with it when they see it

Reply:
Yeah it will show up in the console. I think it's good to keep it because the way to make values persistent (pin debug value to screen) is just RShift, and there is no visual indication whether this action is supported for the selected debug value. So for readonly values pressing RShift will pin the value, and for other values where this is not supported, pressing RShift will print this message to console. This way at least there's at least a chance that user will see the message and understand why nothing happened on screen.